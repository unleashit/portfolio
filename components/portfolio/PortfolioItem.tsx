import React from 'react';
import Link from 'next/link';
import { ASSETS_URL } from '@/lib/constants';
import { DirectusFiles, Project } from '@/services/types/directus';
import { sluggify } from '@/lib/utils';
import styles from './portfolio.module.scss';
import Img from '@/components/common/Img';

export type ProjectWithMeta = Omit<Project, 'image_logo'> & {
  image_logo: DirectusFiles;
};

export default function PortfolioItem({
  item,
}: // index,
// color,
{
  item: ProjectWithMeta;
  index: any;
  color: string;
}) {
  // const color = {
  //   backgroundColor: this.props.color,
  // };

  const {
    image_logo: { width, height },
  } = item;

  return (
    <div className={styles.portfolioItem}>
      <Link href={'/portfolio/' + item.slug}>
        <div className={styles.viewDetails}>
          <span>Learn More</span>
        </div>
        <div>
          <Img
            src={`${ASSETS_URL}/${item.image_logo.id}/${sluggify(
              item.title
            )}.webp?format=webp${width && '&width=' + Math.ceil(+width * 0.8)}`}
            hidpi={`${ASSETS_URL}/${item.image_logo.id}/${sluggify(
              item.title
            )}.webp?format=webp${width && '&width=' + width}`}
            alt={item.title}
            {...(width && { width })}
            {...(height && { height })}
            loading="lazy"
          />

          <div className={styles.portfolioCaption}>
            <p>{item.decription_short}</p>
          </div>
        </div>
      </Link>
    </div>
  );
}
