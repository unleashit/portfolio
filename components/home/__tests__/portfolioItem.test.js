import PortfolioItem from '../portfolioItem';

describe('<PortfolioItem />', () => {
    let wrapper;
    let props = {
        item: {
            url_slug: 'test',
            title: 'title',
            main_image: 'test.jpg',
            description_shot: 'short desc',
        },
        color: '#fff',
    };

    beforeEach(() => {
        wrapper = shallow(<PortfolioItem {...props} />);
    });

    test('renders without crashing', () => {
        expect(wrapper.find('.portfolio-item')).toHaveLength(1);
    });

    test('snapshot matches', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
