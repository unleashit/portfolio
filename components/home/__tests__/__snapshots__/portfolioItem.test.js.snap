// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PortfolioItem /> snapshot matches 1`] = `
<div
  className="portfolio-item"
>
  <Link
    onlyActiveOnIndex={false}
    style={Object {}}
    to="/portfolio/test"
  >
    <div
      className="view-details"
    >
      <span>
        Learn More
      </span>
    </div>
    <div>
      <img
        alt="title"
        src="/images/portfolio/test.jpg"
      />
      <div
        className="portfolio-caption"
      >
        <p />
      </div>
    </div>
  </Link>
</div>
`;
